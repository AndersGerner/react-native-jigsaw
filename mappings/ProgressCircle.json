[{"name":"Progress Circle","tag":"ProgressCircle","description":"A circle used to show completed progress","category":"formControl","preview_image_url":"https://res.cloudinary.com/altos/image/upload/draftbit/library/jigsaw-1.0/reps/Status_Progress.png","supports_list_render":false,"props":{"progress":{"label":"Progress","description":"The amount of progress to display. A number 0-1.","type":"number","value":0.5,"min":0,"max":1,"step":0.01,"precision":2,"editable":true,"required":true},"color":{"label":"Color","description":"Color for the progress shown","editable":true,"value":"primary","required":true,"type":"color"},"unfilledColor":{"label":"Unfilled Color","description":"The color of the unfilled portion of the progress bar(eg. if at 50% then this is the color of the other 50%)","editable":true,"value":null,"required":true,"type":"color"},"fill":{"label":"Inner circle fill","description":"Color of the inner circle","editable":true,"value":null,"required":false,"type":"color"},"size":{"label":"Size","description":"The size of the circle","type":"number","value":100,"min":0,"max":300,"step":1,"precision":1,"editable":true,"required":true},"showsText":{"label":"Show Progress Text","description":"Shows progress in the middle of the circle","type":"boolean","value":true,"editable":true,"required":true},"textStyle":{"label":"Text Style","description":"Text Style of the inner circle","editable":true,"required":true,"type":"typeStyle","value":null},"thickness":{"label":"Thickness","description":"Thickness of the inner circle","type":"number","value":1,"min":0,"max":15,"step":1,"precision":1,"editable":true,"required":true},"direction":{"label":"Direction","description":"Direction that the circle renders in","type":"flatArray","value":"clockwise","options":["counter-clockwise","clockwise"],"editable":true,"required":true},"strokeCap":{"label":"Stroke Cap","description":"The style of the stroke","type":"flatArray","value":"butt","options":["butt","square","round"],"editable":true,"required":true}},"layout":{"width":250,"height":250}}]