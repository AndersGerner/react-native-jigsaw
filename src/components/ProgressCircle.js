import * as React from "react"
import * as Progress from "react-native-progress"
import { COMPONENT_TYPES, FORM_TYPES } from "../core/component-types"
import { withTheme } from "../core/theming"

const ProgressCircle = ({
  progress,
  style,
  color,
  size,
  showsText,
  direction,
  unfilledColor,
  borderColor,
  borderWidth,
  strokeCap,
  fill,
  textStyle,
  thickness
}) => {
  return (
    <Progress.Circle
      progress={progress}
      color={color}
      unfilledColor={unfilledColor}
      borderColor={borderColor}
      borderWidth={borderWidth}
      showsText={showsText}
      direction={direction}
      strokeCap={strokeCap}
      size={size}
      style={style}
      fill={fill}
      textStyle={textStyle}
      thickness={thickness}
    />
  )
}

export default withTheme(ProgressCircle)

export const SEED_DATA = [
  {
    name: "Progress Circle",
    tag: "ProgressCircle",
    description: "A circle used to show completed progress",
    category: COMPONENT_TYPES.formControl,
    preview_image_url: "{CLOUDINARY_URL}/Status_Progress.png",
    supports_list_render: false,
    props: {
      progress: {
        label: "Progress",
        description: "The amount of progress to display. A number 0-1.",
        type: FORM_TYPES.number,
        value: 0.5,
        min: 0,
        max: 1,
        step: 0.01,
        precision: 2,
        editable: true,
        required: true
      },
      color: {
        label: "Color",
        description: "Color for the progress shown",
        editable: true,
        value: "primary",
        required: true,
        type: FORM_TYPES.color
      },
      unfilledColor: {
        label: "Unfilled Color",
        description:
          "The color of the unfilled portion of the progress bar(eg. if at 50% then this is the color of the other 50%)",
        editable: true,
        value: null,
        required: true,
        type: FORM_TYPES.color
      },
      fill: {
        label: "Inner circle fill",
        description: "Color of the inner circle",
        editable: true,
        value: null,
        required: false,
        type: FORM_TYPES.color
      },
      size: {
        label: "Size",
        description: "The size of the circle",
        type: FORM_TYPES.number,
        value: 100,
        min: 0,
        max: 300,
        step: 1,
        precision: 1,
        editable: true,
        required: true
      },
      showsText: {
        label: "Show Progress Text",
        description: "Shows progress in the middle of the circle",
        type: FORM_TYPES.boolean,
        value: true,
        editable: true,
        required: true
      },
      textStyle: {
        label: "Text Style",
        description: "Text Style of the inner circle",
        editable: true,
        required: true,
        type: FORM_TYPES.typeStyle,
        value: null
      },
      thickness: {
        label: "Thickness",
        description: "Thickness of the inner circle",
        type: FORM_TYPES.number,
        value: 1,
        min: 0,
        max: 15,
        step: 1,
        precision: 1,
        editable: true,
        required: true
      },
      direction: {
        label: "Direction",
        description: "Direction that the circle renders in",
        type: FORM_TYPES.flatArray,
        value: "clockwise",
        options: ["counter-clockwise", "clockwise"],
        editable: true,
        required: true
      },
      strokeCap: {
        label: "Stroke Cap",
        description: "The style of the stroke",
        type: FORM_TYPES.flatArray,
        value: "butt",
        options: ["butt", "square", "round"],
        editable: true,
        required: true
      }
    },
    layout: {
      width: 300,
      height: 300
    }
  }
]
