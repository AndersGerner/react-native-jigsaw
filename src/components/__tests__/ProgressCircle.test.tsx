import * as React from "react";
import renderer from "react-test-renderer";
import ProgressCircle from "../ProgressCircle";

it("renders progress circle", () => {
  const tree = renderer.create(<ProgressCircle />).toJSON();

  expect(tree).toMatchSnapshot();
});
