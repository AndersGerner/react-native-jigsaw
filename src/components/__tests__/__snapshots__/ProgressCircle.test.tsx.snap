// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders progress circle 1`] = `
<View>
  <RNSVGSvgView
    bbHeight={100}
    bbWidth={100}
    focusable={false}
    height={100}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        Object {
          "flex": 0,
          "height": 100,
          "width": 100,
        },
      ]
    }
    width={100}
  >
    <RNSVGGroup>
      <RNSVGGroup
        matrix={
          Array [
            1,
            0,
            -0,
            1,
            0,
            0,
          ]
        }
      >
        <RNSVGPath
          d="M 49.96889823477586 0.5000097709080364 A 49.5 49.5 0 1 0 50 0.5"
          fill={0}
          propList={
            Array [
              "fill",
              "stroke",
              "strokeWidth",
              "strokeDasharray",
              "strokeLinecap",
            ]
          }
          stroke={4282108394}
          strokeDasharray={null}
          strokeLinecap={0}
          strokeWidth={1}
        />
      </RNSVGGroup>
    </RNSVGGroup>
  </RNSVGSvgView>
  <View
    style={
      Array [
        Object {
          "alignItems": "center",
          "borderRadius": 49,
          "height": 98,
          "justifyContent": "center",
          "left": 1,
          "overflow": "hidden",
          "position": "absolute",
          "top": 1,
          "width": 98,
        },
        undefined,
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "rgba(90, 69, 255, 1)",
            "fontSize": 27.500000000000004,
          },
          undefined,
        ]
      }
    >
      0
      %
    </Text>
  </View>
</View>
`;
